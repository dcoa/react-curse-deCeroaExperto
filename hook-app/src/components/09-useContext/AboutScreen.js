import React, { useContext } from 'react'
import { UserContext } from './UserContext'


export const AboutScreen = () => {
    const { setUser } = useContext(UserContext)

    return (
        <div>
            <h1>Log out</h1>
            <hr/>
            <button
            className='btn btn-warning'
            onClick={()=>setUser({})}
            >Log out</button>
        </div>
    )
}
