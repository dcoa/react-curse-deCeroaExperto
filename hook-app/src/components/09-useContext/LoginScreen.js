import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext)

    return (
        <div>
            <h1>Log in</h1>
            <hr/>
            <button
            className='btn btn-warning'
            onClick={()=>setUser({
                id: new Date().getTime(),
                name: 'Pikachu',
                type: 'Electric'
            })}
            >Log in</button>
        </div>
    )
}
