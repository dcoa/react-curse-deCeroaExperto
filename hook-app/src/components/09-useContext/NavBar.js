import React from 'react'
import { NavLink } from "react-router-dom";

//NavLink y Link hacen lo mismo la diferencia es que el navlink muestra el activo
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <span class="navbar-text mx-4">
                useContext App
            </span>
            <ul className="navbar-nav">
                <li className="nav-item">
                    <NavLink className="nav-link" to="/">Home</NavLink>
                </li>
                <li className="nav-item">
                    <NavLink className="nav-link" to="/login">Log in</NavLink>
                </li>
                <li className="nav-item">
                    <NavLink className="nav-link" to="/about">About</NavLink>
                </li>
            </ul>
        </nav>
    )
}
