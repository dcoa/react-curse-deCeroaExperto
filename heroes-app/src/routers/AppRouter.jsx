
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { LoginScreen } from "../componets/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
export const AppRouter = () => {
    return (
        <>
            <BrowserRouter>
                <Routes>
                    <Route path="/*" element={
                        <PrivateRoute>
                            <DashboardRoutes />
                        </PrivateRoute>} />
                    {/* <Route path="/*" element={<DashboardRoutes />} /> */}
                    <Route path="/login" element={
                        <PublicRoute>
                            <LoginScreen />
                        </PublicRoute>} />
                    {/* <Route path="/login" element={<LoginScreen />} /> */}
                </Routes>
            </ BrowserRouter>
        </>
    )
}
