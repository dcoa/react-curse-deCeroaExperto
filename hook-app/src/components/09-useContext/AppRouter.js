import React from 'react'
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <BrowserRouter>
            <NavBar />
            <div className='container'>
                <Routes>
                    <Route path="/" exact element={<HomeScreen />} />
                    <Route path="/about" element={<AboutScreen />} />
                    <Route path="/login" element={<LoginScreen />} />
                    <Route path="*" element={<div>404 not found</div>} />
                </Routes>
            </div>
        </BrowserRouter>
    )
}
