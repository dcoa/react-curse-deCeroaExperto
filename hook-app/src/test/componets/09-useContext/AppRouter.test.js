import { mount } from "enzyme"
import { AppRouter } from "../../../components/09-useContext/AppRouter"
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Test on App router', () => {

    const wrapper = mount(
        <UserContext.Provider
            value={{
                user: {
                    name: 'Jhon Doe',
                    email: 'example@fff.com'
                }
            }}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('should show correctly', () => {

        expect(wrapper).toMatchSnapshot();

    });

});
