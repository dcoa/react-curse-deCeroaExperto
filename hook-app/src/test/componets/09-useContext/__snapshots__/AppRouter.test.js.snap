// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on App router should show correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <span
      class="navbar-text mx-4"
    >
      useContext App
    </span>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Log in
      </h1>
      <hr />
      <pre>
        {
   "name": "Jhon Doe",
   "email": "example@fff.com"
}
      </pre>
    </div>
  </div>,
]
`;
